tests/FakeItEasy.Analyzer.Tests/Helpers/DiagnosticResult.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace FakeItEasy.Analyzer.Tests.Helpers
+{
+    using System;
+    using Microsoft.CodeAnalysis;
+
+    /// <summary>
+    /// Location where the diagnostic appears, as determined by path, line number, and column number.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1815:OverrideEqualsAndOperatorEqualsOnValueTypes", Justification = "Not necessary")]
+    public struct DiagnosticResultLocation
+    {
+        public DiagnosticResultLocation(string path, int line, int column)
+        {
+            if (line < 0 && column < 0)
+            {
+                throw new ArgumentException("At least one of line and column must be > 0");
+            }
+
+            if (line < -1 || column < -1)
+            {
+                throw new ArgumentException("Both line and column must be >= -1");
+            }
+
+            this.Path = path;
+            this.Line = line;
+            this.Column = column;
+        }
+
+        public string Path { get; set; }
+
+        public int Line { get; set; }
+
+        public int Column { get; set; }
+    }
+
+    /// <summary>
+    /// Struct that stores information about a Diagnostic appearing in a source.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1815:OverrideEqualsAndOperatorEqualsOnValueTypes", Justification = "Not necessary")]
+    public struct DiagnosticResult
+    {
+        private DiagnosticResultLocation[] locations;
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1819:PropertiesShouldNotReturnArrays", Justification = "It's not intended to be a public API, so it doesn't matter")]

```

Comment:
It's used by `DiagnosticVerifier`, which is public, and has to be public, since it's intended to be the base class for test fixtures (*); IIRC NUnit requires test fixture classes to be public.

(*) admittedly, it's a very questionable design... I guess we could revisit it at some point